ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java

```diff
@@ -462,6 +479,114 @@ public void shouldHandleMaps() {
         assertThat(result, is("value1"));
     }
 
+    @Test
+    public void shouldHandleCaseStatement() {
+        // Given:
+        final Expression expression = analyzeQuery(
+            "SELECT CASE "
+                + "     WHEN col0 < 10 THEN 'small' "
+                + "     WHEN col0 < 100 THEN 'medium' "
+                + "     ELSE 'large' "
+                + "END "
+                + "FROM codegen_test;", metaStore)
+            .getSelectExpressions()
+            .get(0);
+
+        // When:
+        final Object result = codeGenRunner
+            .buildCodeGenFromParseTree(expression, "Case")
+            .evaluate(genericRow(ONE_ROW));
+
+        // Then:
+        assertThat(result, is("small"));
+    }
+
+    @Test
+    public void shouldHandleCaseStatementLazily() {
+        // Given:
+        final Expression expression = analyzeQuery(
+            "SELECT CASE "
+                + "     WHEN WHENCONDITION(true, true) THEN WHENRESULT(100, true) "
+                + "     WHEN WHENCONDITION(true, false) THEN WHENRESULT(200, false) "
+                + "     ELSE WHENRESULT(300, false) "
+                + "END "
+                + "FROM codegen_test;", metaStore)
+            .getSelectExpressions()
+            .get(0);
+
+        // When:
+        final Object result = codeGenRunner
+            .buildCodeGenFromParseTree(expression, "Case")
+            .evaluate(genericRow(ONE_ROW));
+
+        // Then:
+        assertThat(result, is(100));
+    }
+
+    @Test
+    public void shouldOnlyRunElseIfNoMatchInWhen() {
+        // Given:
+        final Expression expression = analyzeQuery(
+            "SELECT CASE "
+                + "     WHEN WHENCONDITION(false, true) THEN WHENRESULT(100, false) "
+                + "     WHEN WHENCONDITION(false, true) THEN WHENRESULT(200, false) "
+                + "     ELSE WHENRESULT(300, true) "
+                + "END "
+                + "FROM codegen_test;", metaStore)
+            .getSelectExpressions()
+            .get(0);
+
+        // When:
+        final Object result = codeGenRunner
+            .buildCodeGenFromParseTree(expression, "Case")
+            .evaluate(genericRow(ONE_ROW));
+
+        // Then:
+        assertThat(result, is(300));
+    }
+
+    @Test
+    public void shouldReturnDefaultForCaseCorrectly() {
```

Comment:
This test doesn’t test anything not already tested in the test above. Consider removing it?

Reply:
Yes, but this one is a bit more targeted to the `ELSE` case. Let's keep it! :)